macos/mac_os_resource_snd.ksy

```diff
@@ -0,0 +1,280 @@
+      sound_header_type:
+        pos: param2 + 20
+        type: u1
+        enum: sound_header_type
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+      sound_header:
+        pos: param2
+        size-eos: true
+        if: is_data_offset and cmd == cmd_type::buffer_cmd
+        type:
+          switch-on: sound_header_type
+          cases:
+            sound_header_type::standard: standard_sound_header
+            sound_header_type::extended: extended_sound_header
+            sound_header_type::compressed: compressed_sound_header
+  standard_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: length
+        type: u4
+        doc: number of samples in array
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: sample_area
+        -orig-id: sampleArea
+        size: length
+        doc: sampled-sound data
+  extended_sound_header:
+    seq:
+      - id: sample_ptr
+        -orig-id: samplePtr
+        type: u4
+        doc: pointer to samples (or 0 if samples follow data structure)
+      - id: num_channels
+        -orig-id: numChannels
+        type: u4
+        doc: number of channels in sample
+      - id: sample_rate
+        -orig-id: sampleRate
+        type: unsigned_fixed_point
+        doc: The rate at which the sample was originally recorded.
+      - id: loop_start
+        -orig-id: loopStart
+        type: u4
+        doc: loop point beginning
+      - id: loop_end
+        -orig-id: loopEnd
+        type: u4
+        doc: loop point ending
+      - id: encode
+        type: u1
+        enum: sound_header_type
+        doc: sample's encoding option
+      - id: base_frequency
+        type: u1
+        doc: base frequency of sample
+        -orig-id: baseFrequency
+      - id: num_frames
+        type: u4
+      - id: aiff_sample_rate
+        size: 10
+        doc: rate of original sample (Extended80)
+      - id: marker_chunk
+        -orig-id: markerChunk
+        type: u4
+        doc: reserved
+      - id: instrument_chunks
+        -orig-id: instrumentChunks
+        type: u4
+        doc: pointer to instrument info
+      - id: aes_recording
+        -orig-id: AESRecording
+        type: u4
+        doc: pointer to audio info
+      - id: sample_size
+        -orig-id: sampleSize
+        type: u2
+        doc: number of bits per sample
```

Comment:
What about `bits_per_sample` _(or `num_bits_per_sample`, but this one is probably overly verbose)_? `size` almost always refers to **byte** size, so using this word as a _number of bits_ is confusing.

`sample_bitsize` would be also a solution, but looks weird to me.